desc/objectives/_free_boundary.py

```diff
@@ -471,12 +484,12 @@ def build(self, use_jit=True, verbose=1):
             source_grid = self._source_grid
 
         if self._eval_grid is None:
-            eval_grid = LinearGrid(
-                rho=np.array([1.0]), M=eq.M_grid, N=eq.N_grid, NFP=eq.NFP, sym=False
-            )
+            eval_grid = source_grid
         else:
             eval_grid = self._eval_grid
 
+        self._eval_grid_is_source_grid = eval_grid == source_grid
```

Comment:
this is basically checking if they are the same object which might be too restrictive, ie if the user passed in `eval_grid=LinearGrid(4,5,6), source_grid=LinearGrid(4,5,6)` this would be False. I think you want `eval_grid.equiv(source_grid)` which will check if they have the same nodes etc.